FBSimulatorControl/Session/FBSimulatorSessionInteraction.m

```diff
@@ -49,15 +50,18 @@ - (instancetype)bootSimulator
   FBSimulatorSessionLifecycle *lifecycle = self.session.lifecycle;
 
   return [self interact:^ BOOL (NSError **error) {
-    NSArray *arguments = @[@"--args",
+    NSMutableArray *arguments = [NSMutableArray arrayWithArray:@[@"--args",
       @"-CurrentDeviceUDID", simulator.udid,
       @"-ConnectHardwareKeyboard", @"0",
-      simulator.configuration.lastScaleCommandLineSwitch, simulator.configuration.scaleString
-    ];
+      simulator.configuration.lastScaleCommandLineSwitch, simulator.configuration.scaleString,
+    ]];
+    if (simulator.pool.configuration.deviceSetPath) {
+      [arguments addObjectsFromArray:@[@"-DeviceSetPath", simulator.pool.configuration.deviceSetPath]];
+    }
 
     id<FBTask> task = [FBTaskExecutor.sharedInstance
       taskWithLaunchPath:simulator.simulatorApplication.binary.path
-      arguments:arguments];
+      arguments:[arguments copy]];
```

Comment:
I don't have particularly strong feels about this but I usually go for immutability if performance isn't a concern.